slave.mk

```diff
@@ -1043,6 +1047,33 @@ $(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : \
 	TARGET_PATH="$(TARGET_PATH)" \
 		./build_image.sh $(LOG)
 
+        # Build an additional image for the DEPENDENT_MACHINE defined.
+	$(foreach dep_machine, $($*_DEPENDENT_MACHINE), \
+		DEBUG_IMG="$(INSTALL_DEBUG_TOOLS)" \
+		DEBUG_SRC_ARCHIVE_FILE="$(DBG_SRC_ARCHIVE_FILE)" \
+		USERNAME="$(USERNAME)" \
+		PASSWORD="$(PASSWORD)" \
+		IMAGE_TYPE=$($*_IMAGE_TYPE) \
+		TARGET_MACHINE=$(dep_machine) \
+		TARGET_PATH=$(TARGET_PATH) \
+		SONIC_ENFORCE_VERSIONS=$(SONIC_ENFORCE_VERSIONS) \
+		TRUSTED_GPG_URLS=$(TRUSTED_GPG_URLS) \
+		PACKAGE_URL_PREFIX=$(PACKAGE_URL_PREFIX) \
+		MULTIARCH_QEMU_ENVIRON=$(MULTIARCH_QEMU_ENVIRON) \
+			./build_debian.sh $(LOG)
+
+		USERNAME="$(USERNAME)" \
+		PASSWORD="$(PASSWORD)" \
+		TARGET_MACHINE=$(dep_machine) \
+		IMAGE_TYPE=$($*_IMAGE_TYPE) \
+		SONIC_ENABLE_IMAGE_SIGNATURE="$(SONIC_ENABLE_IMAGE_SIGNATURE)" \
+		SIGNING_KEY="$(SIGNING_KEY)" \
+		SIGNING_CERT="$(SIGNING_CERT)" \
+		CA_CERT="$(CA_CERT)" \
+		TARGET_PATH="$(TARGET_PATH)" \
+			./build_image.sh $(LOG)
+	)
+
```

Comment:
Looks like we can remove the line 1024~1028.
foreach $($*_DEPENDENT_MACHINE)  and $($*_MACHINE), something like this:
$(foreach dep_machine, $($*_DEPENDENT_MACHINE) $($*_MACHINE), \

Reply:
Updated with the change.